src/Container/ApplicationFactory.php

```diff
@@ -142,9 +143,13 @@ public function __invoke(ContainerInterface $container)
             ? $container->get(RouterInterface::class)
             : new FastRouteRouter();
 
-        $finalHandler = $container->has('Zend\Expressive\FinalHandler')
-            ? $container->get('Zend\Expressive\FinalHandler')
-            : null;
+        if ($container->has('Zend\Expressive\FinalHandler')) {
+            $finalHandler = $container->get('Zend\Expressive\FinalHandler');
+        } else {
+            $config = $container->has('config') ? $container->get('config') : [];
+            $options = isset($config['final_handler']['options']) ? $config['final_handler']['options'] : [];
+            $finalHandler = new FinalHandler($options, null);
+        }
```

Comment:
I'd extract the above logic to a private method, so that the logic here can be simply something like `$finalHandler = $this->marshalFinalHandler($container);`.